servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -456,25 +378,48 @@ public void completeFile(long fileId)
         }
       }
 
-      completeFileInternal(fileInode.getBlockIds(), fileId, fileLength, false, opTimeMs);
+      // If the file is both persisted and stored in memory, the memory footprint and the UFS file
+      // length should match.
+      if (fileInode.isPersisted() && inMemoryLength != 0
+          && options.getUfsLength() != inMemoryLength) {
```

Comment:
I think the problem is that `inMemoryLength` is a bad name. It does not represent that amount of data stored in memory, it represents the amount of data stored in Tachyon. Would renaming it to `tachyonLength` and updating the comment above help?